core/src/main/java/com/microsoft/applicationinsights/channel/concrete/inprocess/InProcessTelemetryChannel.java

```diff
@@ -47,313 +45,339 @@
+	}
+
+	/**
+	 * This Ctor will query the 'namesAndValues' map for data to initialize itself
+	 * It will ignore data that is not of its interest, this Ctor is useful for
+	 * building an instance from configuration
+	 *
+	 * @param namesAndValues
+	 *            - The data passed as name and value pairs
+	 */
+	public InProcessTelemetryChannel(Map<String, String> namesAndValues) {
+		boolean developerMode = false;
+		String endpointAddress = null;
+		int maxInstantRetries = DEFAULT_MAX_INSTANT_RETRY;
+		
+		LimitsEnforcer maxTelemetryBufferCapacityEnforcer = createDefaultMaxTelemetryBufferCapacityEnforcer(null);
+
+		LimitsEnforcer sendIntervalInSecondsEnforcer = createDefaultSendIntervalInSecondsEnforcer(null);
+
+		boolean throttling = true;
+		if (namesAndValues != null) {
+			throttling = Boolean.valueOf(namesAndValues.get("Throttling"));
+			developerMode = Boolean.valueOf(namesAndValues.get(DEVELOPER_MODE_NAME));
+			try {
+				maxInstantRetries = Integer.parseInt(INSTANT_RETRY_NAME);
+			} catch (NumberFormatException e) {
+				InternalLogger.INSTANCE.error("Unable to parse configuration setting %s to integer value.%nStack Trace:%n%s", INSTANT_RETRY_NAME, ExceptionUtils.getStackTrace(e));
+			}
+			
+			if (!developerMode) {
+				developerMode = Boolean.valueOf(System.getProperty(DEVELOPER_MODE_SYSTEM_PROPRETY_NAME));
+			}
+			endpointAddress = namesAndValues.get(ENDPOINT_ADDRESS_NAME);
+
+			maxTelemetryBufferCapacityEnforcer
+					.normalizeStringValue(namesAndValues.get(MAX_MAX_TELEMETRY_BUFFER_CAPACITY_NAME));
+			sendIntervalInSecondsEnforcer
+					.normalizeStringValue(namesAndValues.get(FLUSH_BUFFER_TIMEOUT_IN_SECONDS_NAME));
+		}
+
+		String maxTransmissionStorageCapacity = namesAndValues.get(MAX_TRANSMISSION_STORAGE_CAPACITY_NAME);
+		initialize(endpointAddress, maxTransmissionStorageCapacity, developerMode, maxTelemetryBufferCapacityEnforcer,
+				sendIntervalInSecondsEnforcer, throttling, maxInstantRetries);
+	}
+
+	/**
+	 * Gets value indicating whether this channel is in developer mode.
+	 */
+	@Override
+	public boolean isDeveloperMode() {
+		return developerMode;
+	}
+
+	/**
+	 * Sets value indicating whether this channel is in developer mode.
+	 *
+	 * @param developerMode
+	 *            True or false
+	 */
+	@Override
+	public void setDeveloperMode(boolean developerMode) {
+		if (developerMode != this.developerMode) {
+			this.developerMode = developerMode;
+			int maxTelemetriesInBatch = this.developerMode ? 1 : DEFAULT_MAX_TELEMETRY_BUFFER_CAPACITY;
+
+			setMaxTelemetriesInBatch(maxTelemetriesInBatch);
+		}
+	}
+
+	/**
+	 * Sends a Telemetry instance through the channel.
+	 */
+	@Override
+	public void send(Telemetry telemetry) {
+		Preconditions.checkNotNull(telemetry, "Telemetry item must be non null");
+
+		if (isDeveloperMode()) {
+			telemetry.getContext().getProperties().put("DeveloperMode", "true");
+		}
+
+		if (telemetrySampler != null) {
```

Comment:
Is this just kept for maintaining backward compatibility with Sampling V1? Nonetheless, with this major version we should definitely scrap off the old sampling algorithm as it uses user id based sampling which is not desirable for sure.

Reply:
@dhaval24  - Again, nothing new here. The same code was there before. We can address unrelated changes as separate PR's.